From e889890aaa2aa92a2f6178d34c4f9163f7bd08d8 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 3 Dec 2018 10:40:16 +0800
Subject: [PATCH 10/30] compositor: Support freezing display

Freeze display when the specified file(from env "WESTON_FREEZE_DISPLAY")
exists.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/compositor.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index 9ee5c70..8feb05c 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -2435,6 +2435,9 @@ output_repaint_timer_handler(void *data)
 	void *repaint_data = NULL;
 	int ret;
 
+	if (!access(getenv("WESTON_FREEZE_DISPLAY"), F_OK))
+		goto out;
+
 	weston_compositor_read_presentation_clock(compositor, &now);
 
 	if (compositor->backend->repaint_begin)
@@ -2456,6 +2459,7 @@ output_repaint_timer_handler(void *data)
 						        repaint_data);
 	}
 
+out:
 	output_repaint_timer_arm(compositor);
 
 	return 0;
-- 
2.20.1

